### PR TITLE
Edit/Delete message localization

### PR DESCRIPTION
**Pull request by PorterNan**
# What

Localization for edit + delete messages:
- Add strings into sendbox and messageThread
- Remove some unnecessary variables
# Why

# How Tested

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: packages/react-components/src/localization/locales/en-US/strings.json [hunk 1]
```diff
@@ -60,7 +61,10 @@
     "friday": "Friday",
     "saturday": "Saturday",
     "participantJoined": "joined the chat.",
-    "participantLeft": "left the chat."
+    "participantLeft": "left the chat.",
+    "editMessage": "Edit",
+    "removeMessage": "removeMessage",
```
Comment (JamesBurnside) [inline packages/react-components/src/localization/locales/en-US/strings.json:66]: should the right hand side here be camel case or something like `remove message` or `delete message`?

**Review by JamesBurnside — APPROVED** [review]

File: packages/react-components/src/localization/locales/en-US/strings.json (hunk 1 above)
Comment (PorterNan) [reply packages/react-components/src/localization/locales/en-US/strings.json:66]: What... I think that was a copy&paste problem, thanks for catching it!

File: change/@internal-react-components-db951d2a-5d93-4313-be20-e2336392957b.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "prerelease",
+  "comment": "Localization for edit + delete messages: Add strings into sendbox and messageThread Remove some unnecessary variables",
```
Comment (prprabhu-ms) [inline change/@internal-react-components-db951d2a-5d93-4313-be20-e2336392957b.json:3]: A bit wordy here... :)

File: packages/react-components/src/components/ChatMessageComponent.tsx [hunk 3]
```diff
@@ -86,11 +87,10 @@ const GenerateTextMessageContent = (payload: ChatMessagePayload): JSX.Element =>
 };
 
 export const ChatMessageComponent = (props: ChatMessageProps): JSX.Element => {
-  const strings = useLocale().strings.messageThread;
   const ids = useIdentifiers();
   const theme = useTheme();
 
-  const { message, onUpdateMessage, onDeleteMessage, editDisabled, showDate, messageContainerStyle } = props;
+  const { message, onUpdateMessage, onDeleteMessage, editDisabled, showDate, messageContainerStyle, strings } = props;
```
Comment (prprabhu-ms) [inline packages/react-components/src/components/ChatMessageComponent.tsx:93]: The `strings` field should always be optional and the implementation should fallback to the strings from locale. See `ParticipantItem` for an examples.

File: packages/react-components/src/components/MessageThread.tsx [hunk 4]
```diff
@@ -163,6 +163,12 @@ export interface MessageThreadStrings {
   participantJoined: string;
   /** String for participants left */
   participantLeft: string;
+  /** String for edited tag of a message */
```
Comment (prprabhu-ms) [inline packages/react-components/src/components/MessageThread.tsx:166]: nit: Could improve the description here "Tag shown on a message that has been edited".

File: packages/react-components/src/components/SendBox.tsx [hunk 5]
```diff
@@ -33,6 +32,10 @@ export interface SendBoxStrings {
    * Placeholder text in SendBox when there is no user input
    */
   placeholderText: string;
+  /**
+   * The warning message when send box text length is more than max limit
+   */
+  textExceedsLimit: string;
```
Comment (prprabhu-ms) [inline packages/react-components/src/components/SendBox.tsx:38]: nit: `textTooLong` might be a slightly better name here? Could it be confusing what limit you're talking about? Or is character length the only possible limit here?

File: packages/react-components/src/localization/locales/en-US/strings.json [hunk 6]
```diff
@@ -60,7 +61,10 @@
     "friday": "Friday",
     "saturday": "Saturday",
     "participantJoined": "joined the chat.",
-    "participantLeft": "left the chat."
+    "participantLeft": "left the chat.",
```
Comment (prprabhu-ms) [inline packages/react-components/src/localization/locales/en-US/strings.json:64]: nit: Perhaps a separate PR: These should be sorted.

**Review by prprabhu-ms — APPROVED** [review]
LGTM, pending fallback to locale strings.

**Review by alcail — APPROVED** [review]

**Review by alkwa-msft — APPROVED** [review]

File: packages/react-components/src/components/ChatMessageComponent.tsx (hunk 3 above)
Comment (PorterNan) [reply packages/react-components/src/components/ChatMessageComponent.tsx:93]: This is not a component we are going to expose, this component is just for code/logic separation, so it is always asking for strings from messageThread component
